build-tools/packages/build-cli/src/library/markdown.ts

```diff
@@ -26,18 +27,63 @@ export function remarkHeadingLinks(): (tree: Node) => void {
 			if (node.children?.length > 0) {
 				const firstChild = node.children[0];
 				if (firstChild.type === "text") {
-					const headingValue = firstChild.value;
-					const slug = slugger.slug(headingValue);
+					const slug = slugger.slug(toString(node));
 					// We need to insert an Html node instead of a string, because raw
 					// strings will get markdown-escaped when rendered
 					const htmlNode: Html = {
 						type: "html",
 						value: `<a id="${slug}"></a>`,
 					};
-					// Insert the HTML node before the text node of the heading
+					// Insert the HTML node as the first child node of the heading
 					node.children.unshift(htmlNode);
 				}
 			}
 		});
 	};
 }
+
+/**
+ * A regular expression that extracts an admonition title from a string UNLESS the admonition title is the only thing on
+ * the line.
+ *
+ * Capture group 1 is the admonition type/title. Capture group 2 is any trailing whitespace.
```

Comment:
To prevent it being interpreted as just the `CAUTION`, `NOTE`, etc., text.
```suggestion
 * Capture group 1 is the admonition type/title (from the leading `[!` all the way to the trailing `]`). Capture group 2 is any trailing whitespace.
```